### PR TITLE
🔧 Suggestion for recommended extensions

### LINKED ISSUE #2577: Update recommened extension for vscode
We have quite a few quality of life and mandatory extensions for vscode. Update the recommend extensions.json with latest extensions.

### PR DESCRIPTION
**Pull request by mimarz**
resolves #2577 

Suggested recommended extensions, with some personal recommendations such as rainbow-brackets 🌈 
- [dbaeumer.vscode-eslint](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint)
- [stylelint.vscode-stylelint](https://marketplace.visualstudio.com/items?itemName=stylelint.vscode-stylelint)
- [esbenp.prettier-vscode](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode)
- [editorconfig.editorconfig](https://marketplace.visualstudio.com/items?itemName=editorconfig.editorconfig)
- [redhat.vscode-yaml](https://marketplace.visualstudio.com/items?itemName=redhat.vscode-yaml)
- [unifiedjs.vscode-mdx](https://marketplace.visualstudio.com/items?itemName=unifiedjs.vscode-mdx)
- [deque-systems.vscode-axe-linter](https://marketplace.visualstudio.com/items?itemName=deque-systems.vscode-axe-linter)
- [styled-components.vscode-styled-components](https://marketplace.visualstudio.com/items?itemName=styled-components.vscode-styled-components)
- [aaron-bond.better-comments](https://marketplace.visualstudio.com/items?itemName=aaron-bond.better-comments)
- [seatonjiang.gitmoji-vscode](https://marketplace.visualstudio.com/items?itemName=seatonjiang.gitmoji-vscode)
- [usernamehw.errorlens](https://marketplace.visualstudio.com/items?itemName=usernamehw.errorlens)
- [donjayamanne.githistory](https://marketplace.visualstudio.com/items?itemName=donjayamanne.githistory)
- [ms-vsliveshare.vsliveshare-pack](https://marketplace.visualstudio.com/items?itemName=ms-vsliveshare.vsliveshare-pack)
- [davidanson.vscode-markdownlint](https://marketplace.visualstudio.com/items?itemName=davidanson.vscode-markdownlint)
- [ionutvmi.path-autocomplete](https://marketplace.visualstudio.com/items?itemName=ionutvmi.path-autocomplete)
- [jock.svg](https://marketplace.visualstudio.com/items?itemName=jock.svg)
- [github.vscode-pull-request-github](https://marketplace.visualstudio.com/items?itemName=github.vscode-pull-request-github)

### CODE REVIEW

**Review by martalalik — APPROVED** [review]
Looks good 👍 
Personally i can recommend [Git Graph](https://marketplace.visualstudio.com/items?itemName=mhutchie.git-graph), gives nice overview over brunches, helpful for rebasing.

**Review by nilsml — DISMISSED** [review]
Did you test the extensions with Gitpod?
Got "Couldn't find branch and path for chore/MIM-2577-extensions-recommendations in repo equinor/design-system."

**Comment by mimarz** [comment]
> Did you test the extensions with Gitpod? Got "Couldn't find branch and path for chore/MIM-2577-extensions-recommendations in repo equinor/design-system."

@nilsml How did you test it? 

I tested using the link provided by gitpod in the "checks" on this PR. Works fine.
image

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -34,17 +34,25 @@
...
+    "deque-systems.vscode-axe-linter",
+    "styled-components.vscode-styled-components",
+    "aaron-bond.better-comments",
+    "seatonjiang.gitmoji-vscode",
+    "usernamehw.errorlens",
+    "donjayamanne.githistory",
+    "ms-vsliveshare.vsliveshare-pack",
+    "davidanson.vscode-markdownlint",
+    "ionutvmi.path-autocomplete",
+    "2gua.rainbow-brackets",
```
Comment (oddvernes) [inline .devcontainer/devcontainer.json:52]: I guess remove rainbow since its deprecated in favor of core solution?

**Review by oddvernes — APPROVED** [review]
LGTM

**Pull request merged**